### DISCUSSION TITLE
Does anyone have a working systemd service file for this?

### DISCUSSION THREAD

**Discussion by k0d3g3ar** [AGWA/sms-over-xmpp#44] [General]
Running this on Rocky Linux 8.5 but the service file isn't working.  It works fine if I manually run it as a background task though.

This is what I have.  I suspect it has something to do with the fact that sms-over-xmpp produces output to the console

[Unit]
Description=sms-over-xmpp
After=network.target

[Service]
Type=oneshot
RemainAfterExit=yes
WorkingDirectory=/etc/prosody
ExecStart=/etc/prosody/start-sms-over-xmpp.sh

[Install]
WantedBy=multi-user.target

I'm not an expert in systemd.  Any thoughts?

**Comment by digiospace** [comment]
I know this is late to the punch, but in case anyone is working on setting this up, here is the systemd file that is working for me on a fresh install of Debian Bookworm:
/etc/systemd/system/smsxmpp.service


```
[Unit]
Description=sms-over-xmpp
ConditionPathExists=/opt/smsxmpp/smsxmpp
After=network.target
 
[Service]
Type=simple
Restart=on-failure
RestartSec=10
startLimitIntervalSec=60

WorkingDirectory=/opt/smsxmpp
ExecStart=/opt/smsxmpp/smsxmpp  -listen tcp:8080 -config /opt/smsxmpp
 
[Install]
WantedBy=multi-user.target
```



be sure to do this after creating the file:
`sudo systemctl daemon-reload
`
test with:
`sudo service smsxmpp
`
if it works then do:
`sudo systemctl enable smsxmpp
`
